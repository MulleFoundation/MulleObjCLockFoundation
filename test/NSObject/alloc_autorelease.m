#ifndef __MULLE_OBJC__
# import <Foundation/Foundation.h>
# pragma message "this test does not work with Apple Foundation"
#else
# import <MulleObjC/MulleObjC.h>
#endif


@interface Foo : NSObject
@end

static int   n_instances;

@implementation Foo

+ (id) new
{
   ++n_instances;
   printf( "+new\n");
   return( [super new]);
}

+ (id) alloc
{
   ++n_instances;
   printf( "+alloc\n");
   return( [super alloc]);
}

- (id) init
{
   printf( "-init\n");
   return( [super init]);
}


- (void) finalize
{
   printf( "-finalize\n");
   [super finalize];
}


- (void) dealloc
{
   --n_instances;
   printf( "-dealloc\n");
   [super dealloc];
}

@end


main()
{
   Foo   *foo;

   @autoreleasepool
   {
      foo = [[[Foo alloc] init] autorelease];
      printf( "%d\n", n_instances);
   }
   printf( "%d\n", n_instances);

   return( 0);
}
