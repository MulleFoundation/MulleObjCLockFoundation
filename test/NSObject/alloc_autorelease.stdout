+alloc
-init
1
-finalize
-dealloc
0
